### ISSUE TITLE
Future request: Faster backward- and forward_fill() functions

### ISSUE THREAD

**Issue by Chuck321123** [pola-rs/polars#16875]
### Description

So maybe not the highest priority right now, but I would be happy if we got faster backward- and forward_fill() functions as I think there are more optimization potential to these functions. By running this code: 
```py
import pandas as pd
import numpy as np
import polars as pl

np.random.seed(123)

n_rows = 100_000_000

random_numbers = np.random.rand(n_rows)
nan_mask = np.random.rand(n_rows) < 0.5
random_numbers[nan_mask] = np.nan

# Create DataFrame
df = pd.DataFrame({
    'RandomNumbers': random_numbers
})

print(df.head(10))

df = pl.DataFrame(df)

df = df.with_columns(pl.col("RandomNumbers").fill_nan(None).alias("Results"))

%timeit df.with_columns(pl.col("RandomNumbers").backward_fill().alias("Results"))
``` 
I get these benchmarks: ```1.17 s ± 39.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)``` Obviously, it becomes even slower if you forward and backward fill over groups. Would be nice if someone could find a way to improve these functions.

**Comment by cmdlineluser** [comment]
Just for reference: https://github.com/pola-rs/polars/issues/15480#issuecomment-2129888688

> More improvement with **branchless filling** is possible still but low priority at the moment, as it's rather labour-intensive to write.

**Comment by Chuck321123** [comment]
@cmdlineluser True, thats why I made this open post in case anyone wanted to work on these functions

**Comment by deanm0000** [comment]
did you mean for this to be a future request instead of a feature request?

**Comment by Chuck321123** [comment]
@deanm0000 Yes, my bad. Although, in reality it's an optimization request

**Comment by lukemanley** [comment]
It looks like https://github.com/pola-rs/polars/pull/20689 probably helps here? Maybe closes this issue?

Note comment: https://github.com/pola-rs/polars/issues/20669#issuecomment-2587046275

**Comment by coastalwhite** [comment]
The group aggregation case is addressed now by #25115.
